ttnn/cpp/ttnn/operations/moreh/moreh_dot_op/device/moreh_dot_program_factory.cpp

```diff
@@ -0,0 +1,166 @@
+    uint32_t pad_h = a_shape_wo_padding[2] % TILE_HEIGHT;
+    uint32_t pad_w = a_shape_wo_padding[3] % TILE_WIDTH;
+    uint32_t mask_h = (pad_h == 0) ? (TILE_HEIGHT) : (pad_h);
+    uint32_t mask_w = (pad_w == 0) ? (TILE_WIDTH) : (pad_w);
+
+    tt::tt_metal::Device *device = input_tensor_a.device();
+
+    const uint32_t in0_t = 2;   // a
+    const uint32_t in1_t = 2;   // b
+    const uint32_t in2_t = 1;   // scaler
+    const uint32_t out0_t = 2;  // out
+    const uint32_t im0_t = 1;
+    const uint32_t im1_t = 1;
+
+    CoreCoord core = {0, 0};
+
+
+    tt::operations::primary::CreateCircularBuffer(
+        program,
+        std::set<CoreRange>{CoreRange(core, core)},
+        cb_data_format,
+        {
+            {CB::c_in0, in0_t},
+            {CB::c_in1, in1_t},
+            {CB::c_in2, in2_t},
+            {CB::c_out0, out0_t},
+            {CB::c_intermed0, im0_t},
+            {CB::c_intermed1, im1_t},
+        });
+
+    std::vector<uint32_t> reader_compile_time_args = {
+        (std::uint32_t)tt::operations::primary::is_dram(src0_buffer),
+        (std::uint32_t)tt::operations::primary::is_dram(src1_buffer),
+        *reinterpret_cast<uint32_t *>(&scaler)};
+
+    std::vector<uint32_t> writer_compile_time_args = {(std::uint32_t)CB::c_out0, (std::uint32_t)tt::operations::primary::is_dram(dst_buffer)};
+    const auto reader_kernel_file = "ttnn/cpp/ttnn/operations/moreh/moreh_dot_op/device/kernels/reader_moreh_dot.cpp";
+    const auto writer_kernel_file = "ttnn/cpp/ttnn/operations/moreh/moreh_dot_op/device/kernels/writer_moreh_dot.cpp";
+
+    const auto reader_kernel_id = tt::operations::primary::CreateReadKernel(program, reader_kernel_file, core, reader_compile_time_args);
+    const auto writer_kernel_id = tt::operations::primary::CreateWriteKernel(program, writer_kernel_file, core, writer_compile_time_args);
+
+    vector<uint32_t> compute_kernel_args = {};
+    std::map<string, string> compute_defines;
+    compute_defines["REDUCE_OP"] = "PoolType::SUM";
+    compute_defines["REDUCE_DIM"] = "ReduceDim::REDUCE_ROW";
+
+    const uint32_t core_num = 1;
+    const auto compute_kernel_file = "ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/moreh_dot/single_core/kernels/moreh_dot.cpp";
+    const auto compute_kernel_id =
+        tt::operations::primary::CreateComputeKernel(program, compute_kernel_file, {core, core_num, compute_kernel_args}, compute_defines);
+
+    SetRuntimeArgs(
+        program,
+        reader_kernel_id,
+        core,
+        {src0_buffer->address(), src1_buffer->address(), num_tiles, 0, mask_h, mask_w});
+    SetRuntimeArgs(program, compute_kernel_id, core, {num_tiles, 1});
+    SetRuntimeArgs(program, writer_kernel_id, core, {output_tensor.buffer()->address(), 1, 0});
+
+    const std::vector<Tensor> input_tensors = {input_tensor_a, input_tensor_b};
+
+    auto override_runtime_arguments_callback = [reader_kernel_id, writer_kernel_id, compute_kernel_id](
```

Comment:
Please move this code to `MorehDotOperation::SingleCore::override_runtime_arguments` and delete.